### DISCUSSION TITLE
Buf Proto Breaking: Test Against Branch

### DISCUSSION THREAD

**Discussion by aalobaidi** [rules-proto-grpc/rules_proto_grpc#126] [Q&A] [answered]
Buf CLI supports testing against a git branch or repository URL:

```
$ buf breaking --against '.git#branch=master'
$ buf breaking --against 'https://github.com/foo/bar.git'
```
https://docs.buf.build/breaking-usage

Is there a way to support that in `buf_proto_breaking_test` rule?

Thanks

**Comment by aaliddell** [comment, marked as answer]
We use the [Buf protoc plugin](https://docs.buf.build/breaking-protoc-plugin/)  rather than the executable Buf itself, in order to fit into the workflow of all the other rules here. Unfortunately, they don’t support remote checks with that plugin though:

> "against_input" is required, and is limited to image formats, i.e the format must be bin, json, and cannot be dir, git, tar, zip
